ballerina/listener.bal

```diff
@@ -84,6 +114,12 @@ public class Listener {
         if result is error {
             return error Error("Error occurred while starting the service", result);
         }
+        if self.wsListener is websocket:Listener {
+            result = (<websocket:Listener>self.wsListener).'start();
```

Comment:
@MaryamZi What should be the better way to stop another listener like this?

* To use type-guard and cast
* To define a local variable and then use that

Reply:
The reason why the type is not narrowed for `self.wsListener` is that the value (and therefore the type) may change to be something else from when the check was done vs when the value is actually used in a context where it is expected to be of the tested type. So here for example, we can't guarantee that `self.wsListener` is a `websocket:Listener` in L118. Due to the same reason, using a cast is not generally safe since the `self.wsListener` value may have changed.

Of course, the author may know that this value will never change and may consider it safe. But assigning to a local variable and then doing the type test is guaranteed to be safe.

Moreover, we do the field access just once if we use a local variable, and we can get rid of an unnecessary access.